functests/ptp/ptp.go

```diff
@@ -108,4 +97,472 @@ var _ = Describe("ptp", func() {
+			status = client.Client.ConfigV1Interface.RESTClient().Post().AbsPath(path).
+				Resource("ptpconfigs").Body(ptpConfigSlave.DeepCopy()).Context(context.TODO()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP slave config recovery Error"))
+			err := waitUntilLogIsDetected(testPtpPod, 180, "Profile Name: slave")
+			Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+		})
+	})
 })
+
+func podRole(runningPod v1core.Pod, role string) bool {
+	nodeList, err := client.Client.Nodes().List(metav1.ListOptions{
+		LabelSelector: role,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	for NodeNumber := range nodeList.Items {
+		if runningPod.Spec.NodeName == nodeList.Items[NodeNumber].Name {
+			return true
+		}
+	}
+	return false
+}
+
+func execCommand(pod v1core.Pod, command []string) (string, bytes.Buffer) {
+	req := client.Client.CoreV1Interface.RESTClient().
+		Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		VersionedParams(&v1core.PodExecOptions{
+			Container: pod.Spec.Containers[0].Name,
+			Command:   command,
+			Stdin:     true,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       true,
+		}, scheme.ParameterCodec)
+	retry := 3
+	for retry > 0 {
+		exec, err := remotecommand.NewSPDYExecutor(client.Client.Config, "POST", req.URL())
+		Expect(err).ToNot(HaveOccurred())
+		var buf bytes.Buffer
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  os.Stdin,
+			Stdout: &buf,
+			Stderr: os.Stderr,
+			Tty:    true,
+		})
+		if err == nil {
+			commandOut := buf.String()
+			return commandOut, buf
+		}
+		log.Print(err)
+		retry--
+		time.Sleep(time.Second)
+	}
+	return "", bytes.Buffer{}
+}
+
+func compareOffsetTime(timeDiff string) bool {
+	var timeStampList []int
+	for _, line := range strings.Split(timeDiff, "\n") {
+		if strings.Contains(line, "openshift_ptp_max_offset_from_master") && !strings.Contains(line, "# ") {
+			splitedLine := strings.Split(line, " ")
+			offcetFromMaster, err := strconv.Atoi(strings.Trim(splitedLine[len(splitedLine)-1], "\r"))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(offcetFromMaster).To(BeNumerically("<", 100))
+			timeStampList = append(timeStampList, offcetFromMaster)
+		}
+	}
+	Expect(len(timeStampList)).To(BeNumerically("==", 2))
+	return true
+}
+
+func getPodLogs(pod v1core.Pod) (string, error) {
+	podLogOpts := v1core.PodLogOptions{}
+	req := client.Client.CoreV1Interface.Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream()
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	str := buf.String()
+
+	return str, nil
+}
+
+func netParseEthtoolTimeStampFeature(cmdOut *bytes.Buffer) bool {
+	var hardRxEnabled bool
+	var hardTxEnabled bool
+	var hardRawEnabled bool
+
+	glog.V(2).Infof("cmd output for %v", cmdOut)
```

Comment:
why we need this glog?

Reply:
That's what happened when you copy/paste code :) . Fixed